src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -110,12 +113,35 @@ private CaseSearchResult toCaseDetailsSearchResult(MultiSearchResult multiSearch
                 throw new BadSearchRequest(errMsg);
             }
             if (response.searchResult != null) {
+                buildCaseFieldsAggregations(response, caseFieldsAggregations, crossCaseTypeSearchRequest);
                 caseDetails.addAll(searchResultToCaseList(response.searchResult));
                 totalHits += response.searchResult.getTotal();
             }
         }
 
-        return new CaseSearchResult(totalHits, caseDetails);
+        return new CaseSearchResult(caseFieldsAggregations, totalHits, caseDetails);
+    }
+
+    private void buildCaseFieldsAggregations(
+        final MultiSearchResult.MultiSearchResponse response,
+        final List<CaseTypesResults> caseFieldsAggregations,
+        final CrossCaseTypeSearchRequest crossCaseTypeSearchRequest) {
+
+        if (response.searchResult.getJsonObject().getAsJsonObject("hits").get("hits").getAsJsonArray().size() != 0) {
+
+            final String indexName = response.searchResult.getJsonObject().getAsJsonObject("hits").get("hits")
+                .getAsJsonArray().get(0).getAsJsonObject().get("_index").toString();
```

Comment:
this code is difficult to read and quickly understand. Please add an utility method with a nice name in MultiSearchResult.MultiSearchResponse

Reply:
can we call it `hitsIsNotEmpty` please

Reply:
As per conversation MultiSearchResult.MultiSearchResponse is part of the framework. However hitsIsNotEmpty() method has been created.  DONE